rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -618,81 +618,111 @@ rmw_ret_t rmw_init()
   return RMW_RET_OK;
 }
 
-rmw_node_t * rmw_create_node(const char * name, size_t domain_id)
+rmw_node_t * rmw_create_node(const char * name, const char * namespace_, size_t domain_id)
 {
   if (!name) {
     RMW_SET_ERROR_MSG("name is null");
     return NULL;
   }
 
+  if (!namespace_) {
+    RMW_SET_ERROR_MSG("namespace_ is null");
+    return NULL;
+  }
+
   eprosima::fastrtps::Log::SetVerbosity(eprosima::fastrtps::Log::Error);
 
+  // Declare everything before beginning to create things.
+  Participant * participant = nullptr;
+  rmw_guard_condition_t * graph_guard_condition = nullptr;
+  CustomParticipantInfo * node_impl = nullptr;
+  rmw_node_t * node_handle = nullptr;
+  topicnamesandtypesReaderListener * tnat_1 = nullptr;
+  topicnamesandtypesReaderListener * tnat_2 = nullptr;
+  std::pair<StatefulReader *, StatefulReader *> EDPReaders;
```

Comment:
Still the same below: https://github.com/ros2/rmw_fastrtps/pull/97/files#diff-8225b098f1ec4debcdcee03d65ae95d5R752